scripts/sct_compute_mtr.py

```diff
@@ -56,52 +56,37 @@ def main(args=None):
 
     fname_mt0 = arguments['-mt0']
     fname_mt1 = arguments['-mt1']
-    remove_temp_files = int(arguments['-r'])
+    fname_mtr = arguments['-o']
+    file_out = os.path.basename(fname_mtr)[:os.path.basename(fname_mtr).rfind('.nii')]
+    ext_out = os.path.basename(fname_mtr)[os.path.basename(fname_mtr).rfind('.nii'):]
     verbose = int(arguments.get('-v'))
     sct.init_sct(log_level=verbose, update=True)  # Update log level
 
-    # Extract path/file/extension
-    path_mt0, file_mt0, ext_mt0 = sct.extract_fname(fname_mt0)
-    path_out, file_out, ext_out = '', file_out, ext_mt0
-
-    # create temporary folder
-    path_tmp = sct.tmp_create()
-
-    # Copying input data to tmp folder and convert to nii
-    sct.printv('\nCopying input data to tmp folder and convert to nii...', verbose)
-    from sct_convert import convert
-    convert(fname_mt0, os.path.join(path_tmp, "mt0.nii"), dtype=np.float32)
-    convert(fname_mt1, os.path.join(path_tmp, "mt1.nii"), dtype=np.float32)
-
-    # go to tmp folder
-    curdir = os.getcwd()
-    os.chdir(path_tmp)
-
     # compute MTR
     sct.printv('\nCompute MTR...', verbose)
-    nii_mt1 = msct_image.Image('mt1.nii')
+    nii_mt1 = msct_image.Image(fname_mt1)
     data_mt1 = nii_mt1.data
-    data_mt0 = msct_image.Image('mt0.nii').data
+    data_mt0 = msct_image.Image(fname_mt0).data
     data_mtr = 100 * (data_mt0 - data_mt1) / data_mt0
     # save MTR file
     nii_mtr = nii_mt1
     nii_mtr.data = data_mtr
-    nii_mtr.save("mtr.nii")
-    # sct.run(fsloutput+'fslmaths -dt double mt0.nii -sub mt1.nii -mul 100 -div mt0.nii -thr 0 -uthr 100 mtr.nii', verbose)
+    nii_mtr.save(fname_mtr)
+    # sct.run(fsloutput+'fslmaths -dt double mt0.nii -sub mt1.nii -mul 100 -div mt0.nii -thr 0 -uthr 100 file_out.nii', verbose)
 
-    # come back
-    os.chdir(curdir)
+    # if changing output file name or location, create folder with mt0.nii.gz and mt1.nii.gz files at precised location
```

Comment:
why do you need that?

Reply:
if we don't copy the mt0 and mt1 files, the command to visualize the output file with fsleyes does not work

Reply:
> the command to visualize the output file with fsleyes does not work

why doesn't it work? the files are physically there

Reply:
The command works when outputting the file into the default folder (the same one that physically holds the mt1 and mt0 files) but if changing the output file location (i.e. Desktop) and the mt1 and mt0 files are not there, you get the following terminal log:
```
$ fsleyes mt0.nii.gz mt1.nii.gz mtr.nii.gz &
[1] 3362
$ unrecognized arguments: mt0.nii.gz mt1.nii.gz
FSLeyes version 0.26.1
Usage: fsleyes [options] file [displayOpts] file [displayOpts] ...
```

Reply:
when the user launches `sct_compute_mtr`, the output location of `fname_mtr` is relative to where the user is located when launching the command.

Reply:
I revised the script and the problem was in the way I wrote the code. With the way I wrote it, the command could not be issued from the folder the user is located. However, with your latest suggestion (changing file_out + ext_out to fname_mtr), the command can now be issued from the current folder.

Changes can be viewed in commit [25defd6](https://github.com/YangHee-Min/spinalcordtoolbox/commit/25defd618e92c464100eee81949aafddf4f8cbfd)